mantidproject/mantid
# D2B loader and IDF updates

Loader for D2B, along with updates for the D2B IDF and some bug fixes in `ScanningWorkspaceBuilder`.

`LoadILLDiffraction` should now build a detector scan workspace for D2B data, an example file for this has been added.

The D2B IDF has some corrections made to it, which have been verified with instrument scientists. The generator for this has been added to mantidgeometry, and should be merged when this PR is approved.

https://github.com/mantidproject/mantidgeometry/compare/19750_D2B_IDF_Generator

The changes to the D2B IDF will mean `LoadILLASCII` is no longer supported for D2B. It is not safe to use in any case, and will be deprecated in a separate PR.

**To test:**

Find the test data for D2B in `ILL/D2B/508093.nxs`. Load this file, and make some checks against the NeXus file to make sure there are no obvious issues. Check documentation reads ok.

Code review.

Fixes #19750.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
